Python/ast.c

```diff
@@ -1666,6 +1703,34 @@ ast_for_decorated(struct compiling *c, const node *n)
     return thing;
 }
 
+static expr_ty
+ast_for_namedexpr(struct compiling *c, const node *n)
+{
+    /* if_stmt: 'if' namedexpr_test ':' suite ('elif' namedexpr_test ':' suite)*
+         ['else' ':' suite]
+       namedexpr_test: test [':=' test]
+       argument: ( test [comp_for] |
+            test ':=' test |
+            test '=' test |
+            '**' test |
+            '*' test )
+    */
+    expr_ty target, value;
+
+    target = ast_for_expr(c, CHILD(n, 0));
+    if (!target)
+        return NULL;
+
+    value = ast_for_expr(c, CHILD(n, 2));
+    if (!value)
+        return NULL;
+
+    if(!set_context(c, target, NamedStore, n))
```

Comment:
PEP-7: Missed space before `(`. Also add braces around `if`s body, even if it is just `return`.